### ISSUE TITLE
Error in test framework outside of repo root directory

### ISSUE THREAD

**Issue by quartox** [python/mypy#3412]
If you run `py.test` outside of the root directory of the repo you get a temporary file not found error with a long stacktrace.

The key part of the message message:

```
suffix = '', prefix = 'mypy-test-'
dir = '/home/jlord/work/mypy/mypy/test/tmp-test-dirs'

    def mkdtemp(suffix=None, prefix=None, dir=None):
        """User-callable function to create and return a unique temporary
        directory.  The return value is the pathname of the directory.
    
        Arguments are as for mkstemp, except that the 'text' argument is
        not accepted.
    
        The directory is readable, writable, and searchable only by the
        creating user.
    
        Caller is responsible for deleting the directory when done with it.
        """
    
        prefix, suffix, dir, output_type = _sanitize_params(prefix, suffix, dir)
    
        names = _get_candidate_names()
        if output_type is bytes:
            names = map(_os.fsencode, names)
    
        for seq in range(TMP_MAX):
            name = next(names)
            file = _os.path.join(dir, prefix + name + suffix)
            try:
>               _os.mkdir(file, 0o700)
E               FileNotFoundError: [Errno 2] No such file or directory: '/home/jlord/work/mypy/mypy/test/tmp-test-dirs/mypy-test-y59peay8'

/home/jlord/miniconda3/envs/mypy/lib/python3.6/tempfile.py:368: FileNotFoundError
```

**Comment by gvanrossum** [comment]
Honestly that falls in the category "don't do that"...

**Comment by gnprice** [comment]
I'd be totally happy with this not working, but it's not good for it to produce a huge error trace like it does. Especially for someone who's just getting started with the project, it's hard to tell whether it's caused by a change they've just made in the code, or their environment isn't set up properly to run our tests at all, or they're doing something wrong in running them and if so what.

Fortunately I think this is easy to fix -- pytest already knows what the root of the tree is (thanks to spotting our `pytest.ini`), so it's just a matter of our own code avoiding accidental dependencies on what the current directory is. I'll take a look.

**Comment by gnprice** [comment]
Fixed in 45a5cf32e.
